hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,270 @@
+In this mode, the hardware peripherals request more data from the EDN by asserting the `req` signal.
+When random values are available, the EDN transmits them on the `bus` and asserts an `ack` signal to signify the arrival of fresh values.
+
+Alternatively peripheral port 0 can be configured in broadcast mode.
+When peripheral port 0 is in broadcast mode, the EDN block will make a CSRNG request for that port without requiring a peripheral module request.
+Once the `genbits` data is returned to the EDN block from CSRNG, it forwards the response to the peripheral port 0 data bus.
+A configurable timer will determine the rate at which `genbits` data will be delivered on the broadcast bus.
+Several peripherals are allowed to connect to the broadcast bus.
+Each individual peripheral will decide whether to take the data when the ack signal is active.
+
+Application interface commands to the CSRNG block can be generated by either firmware or hardware.
+
+Firmware can issue CSRNG commands on behalf of hardware peripherals, by writing the them to the {{< regref "SW_CMD_REQ" >}} register and the command status response is captured in the {{< regref "SW_CMD_STS" >}} register.
+Even when CRSNG `generate` commands are issued by firmware, all random values are distributed to the hardware peripherals.
+
+If firmware applications require random values for their *own* use, they must issue the commands directly to the CSRNG, which maintains a dedicated CSRNG instance for firmware that is accessible through TL-UL.
+
+There are two modes for EDN hardware to generate CSRNG commands.
+One is the "auto request mode", where two FIFOs are used to send commands.
+The general operation of this mode is that the CSRNG instance is set up by firmware, then the FIFOs are preloaded with commands.
+One FIFO can be programmed to send `generate` commands.
+The other FIFO can be programmed to send `reseed` commands.
+The {{< regref "MAX_NUM_REQS_BETWEEN_RESEEDS" >}} register sets the number of `generate` commands allowed between `reseed` commands.
+Once this is done, the EDN block can request data from the CSRNG once firmware has instantiated the associated instance through the EDN command forwarding interface.
+Once in this mode, the EDN emits `generate` commands from the first FIFO to get more data.
+Once the {{< regref "MAX_NUM_REQS_BETWEEN_RESEEDS" >}} timer expires, the EDN block emits a `reseed` command from the second FIFO.
+The process of sending these two commands will repeat forever until the `EDN_ENABLE` bit is cleared, the `AUTO_REQ_MODE` bit is cleared, or the EDN is reset.
+
+The other mode is "boot-time request mode", where only the hardware generates CSRNG application interface commands.
+A single `instantiate` command is sent, followed by a stream of `generate` commands.
+These generate commands continue until `EDN_ENABLE` bit is cleared, the `BOOT_REQ_DIS` bit is set, or the EDN is reset, at which point an `uninstantiate` command is sent to destroy the instance.
```

Comment:
At the start of time we are all Hydrogen and sparse amounts of Helium & Lithium.  Big Bang nucleosynthesis was pretty pathetic indeed, so all this conversation about Silicon is irrelevant.   Though in all seriousness it's a good point.  Will come back to this....

Reply:
So I see your point.  Reset cannot send an uninstantiate.  However, that also means that EDN should never be reset without resetting CSRNG, because that would leave stale state.

Reply:
The text will read as follows in the next commit: 
```
The other mode is "boot-time request mode", where only the hardware generates CSRNG application interface commands.
In this mode a single `instantiate` command is sent, followed by a stream of `generate` commands.
This sequence fetches the initial random values needed for the system to boot.
Use of boot-time request mode, though simpler in operation, is only for applications which do not require FIPS-approved random values.
Please see the [entropy_src IP documentation]({{< relref "hw/ip/entropy_src/doc" >}}) for more information on trade-offs when creating CSRNG seeds before the completion of the FIPS-required health checks.
In boot-time request mode the generate commands continue until `EDN_ENABLE` bit is cleared, the `BOOT_REQ_DIS` bit is set, or the EDN is reset.
When the `EDN_ENABLE` bit is cleared or the `BOOT_REQ_DIS` bit is set an `uninstantiate` command is sent to destroy the instance.
Note that the EDNs and CSRNGs should always be reset together to ensure proper.instantiation or uninstantion of state varibles.